include/cliffordsynthesis/encoding/GateEncoder.hpp

```diff
@@ -30,18 +30,24 @@
     logicbase::LogicMatrix3D gS{};
     // variables for the two-qubit gates
     logicbase::LogicMatrix3D gC{};
+    // variables for the pauli-qubit gates
+    logicbase::LogicMatrix gP{};
 
     void
          collectSingleQubitGateVariables(std::size_t pos, std::size_t qubit,
                                          logicbase::LogicVector& variables) const;
     void collectTwoQubitGateVariables(std::size_t pos, std::size_t qubit,
                                       bool                    target,
                                       logicbase::LogicVector& variables) const;
+    void
+    collectPauliQubitGateVariables(std::size_t             qubit,
```

Comment:
This method is only used by the TwoQubitGateEncoder, please move the method there.